lib/identity/hostdata.rb

```diff
@@ -69,14 +71,23 @@ def self.in_datacenter
       yield self if in_datacenter?
     end
 
+    # @return [String]
+    def self.aws_region
+      @aws_region ||= ENV['LOGIN_AWS_REGION'] ||  Identity::Hostdata::EC2.load.region
```

Comment:
let's cache an ec2 instance because EC2.load is an HTTP request. And then use the same pattern of memoizing even a nil value

```
def self.ec2
  @ec2 ||= EC2.load
end
```